// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.chat";
option java_outer_classname = "Chat";
option objc_class_prefix = "HLW";

package chat;

service ServiceDiscovery {
  rpc RegisterClient (RegisterReq) returns (RegistrationReply) {}
  rpc DeregisterClient (DeregisterReq) returns (RegistrationReply) {}
  rpc GetRegisteredClients (GetRegisteredClientsReq) returns (RegisteredClientsReply) {}
}

message RegisterReq {
  string name = 1;
}

message DeregisterReq {
	string name = 1;
}

message GetRegisteredClientsReq {
}

message RegisteredClientsReply {
  repeated Client client = 1;
}

message Client {
  string name = 1;
  string full_address = 2;
}

message RegistrationReply {
  bool status = 1;
  string full_address = 2;
}

service Chat {
  rpc SendMsg (MessageReq) returns (Ack) {}
}

message MessageReq {
  string message = 1;
  string name = 2;
}

message Ack {
  
}